src/core/reactor_backend.cc

```diff
@@ -131,7 +134,17 @@ aio_storage_context::~aio_storage_context() {
 }
 
 future<> aio_storage_context::stop() noexcept {
+    // Prevent new retries from starting during shutdown
+    _stopping = true;
+
+    // Drain io_sink and complete all pending IOs with error
+    _r._io_sink.drain([] (const internal::io_request& req, io_completion* desc) -> bool {
+        desc->complete_with(-ECANCELED);
+        return true;
+    });
+
```

Comment:
Q: why are we even here? If we're shutting down the reactor, nothing should be in progress.